src/features/post/model/usePostFilter.ts

```diff
@@ -0,0 +1,79 @@
+import { useLocation, useNavigate } from "react-router-dom"
+import { PostFilterParams_i } from "./types"
+import { useEffect, useState } from "react"
+
+interface UsePostFilterProps_i {
+  onFilterChange?: (params: PostFilterParams_i) => void
+}
+
+export const usePostFilter = ({ onFilterChange }: UsePostFilterProps_i) => {
```

Comment:
> 개인적으로 함수의 파라미터가 3개 이상일 경우 객체로 관리하는 것이 편하다고 생각하는데요..!
객체 내 값이 모두 옵셔널일 경우에 인자로 아래와 같이 {}를 넘기는 경우가 생기는데,
이건 제 설계의 문제일까요..? 어떻게 개선하면 좋을지 알고 싶습니다..!

> const { searchQuery } = usePostFilter({})

```suggestion
export const usePostFilter = ({ onFilterChange }?: UsePostFilterProps_i) => {
```

요렇게 하면 객체를 안넘겨도 될것같네요..!
의도하신 질문이 맞을까요?